src/assets/components/PokemonTable/PokemonTable.tsx

```diff
@@ -0,0 +1,29 @@
+import { fetchPokemons } from "./mockData";
+import DataGridTable from "../shared/DataGridTable/DataGridTable";
+import { pokemonCols, PokemonType } from "./types";
+import { useQuery } from "@tanstack/react-query";
+
+function PokemonTable() {
+  const { data, isLoading, isError } = useQuery<PokemonType[]>({
+    queryKey: ["pokemons"],
+    queryFn: fetchPokemons, 
+  });
+  const handleRowClick = (row: { [key: string]: any }) => {
+    console.log("Row clicked:", row);
+  };
+
+  return (
+    <DataGridTable
+      columns={pokemonCols}
+      rows={data || []}
```

Comment:
Instead of this
```suggestion
      rows={data || []}
```

use early return
if(!data.length) return <CircularProgress />